Return-Path: <linux-wireless-owner@vger.kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.3.1 (2010-03-16) on dual
X-Spam-Level: 
X-Spam-Status: No, score=0.9 required=5.0 tests=BAYES_00,FREEMAIL_FROM,
	KB_DATE_CONTAINS_TAB autolearn=no version=3.3.1
X-Original-To: maks@dual
Delivered-To: maks@dual
Received: from dual (localhost.localdomain [127.0.0.1])
	by dual (Postfix) with ESMTP id 4027E2429D
	for <maks@dual>; Thu, 15 Apr 2010 14:57:47 +0200 (CEST)
X-Original-To: max@stro.at
Delivered-To: max@stro.at
Received: from baikonur.stro.at [213.239.196.228]
	by dual with POP3 (fetchmail-6.3.13)
	for <maks@dual> (single-drop); Thu, 15 Apr 2010 14:57:47 +0200 (CEST)
Received: from vger.kernel.org (vger.kernel.org [209.132.180.67])
	by baikonur.stro.at (Postfix) with ESMTP id 785425C00F
	for <max@stro.at>; Thu, 15 Apr 2010 14:09:33 +0200 (CEST)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
	id S1752356Ab0DOMRR (ORCPT <rfc822;max@stro.at>);
	Thu, 15 Apr 2010 08:17:17 -0400
Received: from mail-ew0-f220.google.com ([209.85.219.220]:35717 "EHLO
	mail-ew0-f220.google.com" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
	with ESMTP id S1752038Ab0DOMRQ (ORCPT
	<rfc822;linux-wireless@vger.kernel.org>);
	Thu, 15 Apr 2010 08:17:16 -0400
Received: by ewy20 with SMTP id 20so459287ewy.1
        for <linux-wireless@vger.kernel.org>; Thu, 15 Apr 2010 05:17:15 -0700 (PDT)
DKIM-Signature:	v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=googlemail.com; s=gamma;
        h=domainkey-signature:received:received:received:received:received
         :from:to:subject:date:user-agent:cc:mime-version:content-type
         :content-transfer-encoding:message-id;
        bh=p4STqoMJiC6z/cKYzv2Exm9RYHj8L3PhiKgV8V/My+8=;
        b=Df3+ZF5p9OAOZbcMmdGwwyRhe6KKgy9XnEhFVlpwzN2ifr78NiUNl9AwiYsHd2bCAb
         DFIq9aj2Lw3AzsJziQaRZjAyi1BCwRHWYVT1z6DAUac3GKAPDbTwg8ci9O31eyKBcbQu
         3d//ULPy6g8NH7caloyA+ZhU8cmIl93ddMv0I=
DomainKey-Signature: a=rsa-sha1; c=nofws;
        d=googlemail.com; s=gamma;
        h=from:to:subject:date:user-agent:cc:mime-version:content-type
         :content-transfer-encoding:message-id;
        b=n2N7kl6h8TV/cyVhTSMxV505sNe8/00nKdlLKyTn7h0M9MPZhLYgvJms0gr6wMtuoS
         t78oiQsjdGsh2YMGVes/1eMoISXolb2pGF36Lp3UnoVJtJx5s6aBasGQ0oVUc2PIYozG
         FyMOuuxhdfOyNOE8iJVAGWVBWLNxyIICBR4tU=
Received: by 10.213.2.81 with SMTP id 17mr25407ebi.76.1271333835129;
        Thu, 15 Apr 2010 05:17:15 -0700 (PDT)
Received: from blech.mobile (nat-wh.rz.uni-karlsruhe.de [129.13.72.197])
        by mx.google.com with ESMTPS id 15sm922586ewy.12.2010.04.15.05.17.14
        (version=TLSv1/SSLv3 cipher=RC4-MD5);
        Thu, 15 Apr 2010 05:17:14 -0700 (PDT)
Received: from blech.mobile ([127.0.0.1])
	by localhost (localhost [127.0.0.1]) (amavisd-new, port 10024)
	with ESMTP id gSWm7mhNvXD9; Thu, 15 Apr 2010 14:17:08 +0200 (CEST)
Received: from blech.mobile (localhost [127.0.0.1])
	by blech.mobile (Postfix) with ESMTP id 6D2D834077B;
	Thu, 15 Apr 2010 14:17:08 +0200 (CEST)
From: Christian Lamparter <chunkeey@googlemail.com>
To: linux-wireless@vger.kernel.org
Subject: [PATCH] p54pci: fix serious sparse warning
Date:	Thu, 15 Apr 2010 14:17:07 +0200
User-Agent: KMail/1.12.4 (Linux/2.6.34-rc3-uber-wl; KDE/4.3.4; x86_64; ; )
Cc: linville@tuxdriver.com
MIME-Version: 1.0
Content-Type: Text/Plain;
  charset="iso-8859-1"
Content-Transfer-Encoding: 7bit
Message-Id: <201004151417.07538.chunkeey@googlemail.com>
Sender: linux-wireless-owner@vger.kernel.org
Precedence: bulk
List-ID: <linux-wireless.vger.kernel.org>
X-Mailing-List:	linux-wireless@vger.kernel.org
X-Virus-Scanned: by Amavis (ClamAV) at stro.at
Content-Length: 1202

This patch fixes a bug which was just recently introduced by
("p54pci: prevent stuck rx-ring on slow system").

make M=drivers/net/wireless/p54 C=2 CF=-D__CHECK_ENDIAN__
  CHECK   drivers/net/wireless/p54/p54pci.c                                                                 
drivers/net/wireless/p54/p54pci.c:143:11: warning: cast to restricted __le32                                
  CC [M]  drivers/net/wireless/p54/p54pci.o

Reported-by: Johannes Berg <johannes@sipsolutions.net>
Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
---
diff --git a/drivers/net/wireless/p54/p54pci.c b/drivers/net/wireless/p54/p54pci.c
index 86f3e9a..679da7e 100644
--- a/drivers/net/wireless/p54/p54pci.c
+++ b/drivers/net/wireless/p54/p54pci.c
@@ -140,7 +140,7 @@ static void p54p_refill_rx_ring(struct ieee80211_hw *dev,
 
 	idx = le32_to_cpu(ring_control->host_idx[ring_index]);
 	limit = idx;
-	limit -= le32_to_cpu(index);
+	limit -= index;
 	limit = ring_limit - limit;
 
 	i = idx % ring_limit;
--
To unsubscribe from this list: send the line "unsubscribe linux-wireless" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

